### PR TITLE
use MutationObserver to fix the footer issue

### PR DESCRIPTION
**Pull request by thutterer**
As all other ways to fix the footer issues have downsides, it's time to bring the big guns.

### CODE REVIEW

File: hawk/app/assets/javascripts/application.js [hunk 1]
```diff
@@ -54,14 +52,24 @@ $(function() {
     );
   };
 
+  var createMutationObserver = function() {
+    var target = document.querySelector('#middle');
+    var observer = new MutationObserver(function(mutations) {
+      observer.disconnect();
```
Comment (tboerger) [inline hawk/app/assets/javascripts/application.js]: You are accessing `observer` which gets defined outside with the? Looks wired...

**Comment by tboerger** [comment]
The resize method only increases the size but don't decrease it :)

**Comment by thutterer** [comment]
> The resize method only increases the size but don't decrease it

Maybe I misunderstand, but I think it does what it should.
E.g. removing rules from a location constraint does bring the footer back up.

**Comment by tboerger** [comment]
For the last try from my side it only increased the size

**Pull request merged**